Sources/joyfill-components-swift/Components/ComponentViewController/APIComponents/JoyDoc.swift

```diff
@@ -0,0 +1,466 @@
+        case type, targetValue, fontSize, fontColor, fontStyle, fontWeight, textAlign, primaryDisplayOnly, format, column
+    }
+}
+
+// MARK: - View
+struct View: Codable {
+    let type: String?
+    let pageOrder: [String]?
+    let pages: [Page]?
+    let id: String?
+    
+    enum CodingKeys: String, CodingKey {
+        case type, pageOrder, pages
+        case id = "_id"
+    }
+}
+
+// MARK: - Extension to fetch data from model
+public var jsonData = Data()
+extension JoyDoc {
+    public static func loadFromJSON() -> [JoyDoc]? {
+        do {
+            joyDocStruct = try JSONDecoder().decode(JoyDoc.self, from: jsonData)
+            
+            // It will prevent tasks to perform on main thread
+            DispatchQueue.main.async {
+                // Deinitialize arrays to protect memory leakage
+                cellView.removeAll()
+                yPointsData.removeAll()
+                xPointsData.removeAll()
+                optionsData.removeAll()
+                componentId.removeAll()
+                signedImage.removeAll()
+                yCoordinates.removeAll()
+                xCoordinates.removeAll()
+                joyDocPageId.removeAll()
+                chartPointsId.removeAll()
+                richTextValue.removeAll()
+                tableRowOrder.removeAll()
+                componentType.removeAll()
+                graphLabelData.removeAll()
+                tableCellsData.removeAll()
+                dropdownOptions.removeAll()
+                tableColumnType.removeAll()
+                joyDocFieldData.removeAll()
+                tableFieldValue.removeAll()
+                tableColumnTitle.removeAll()
+                chartValueElement.removeAll()
+                tableColumnOrderId.removeAll()
+                multiSelectOptions.removeAll()
+                multiSelectOptionId.removeAll()
+                joyDocFieldPositionData.removeAll()
+                componentTableViewCellHeight.removeAll()
+                multiChoiseSelectedIndexPath.removeAll()
+                componentsYValueForMobileView.removeAll()
+                selectedDropdownOptionIndexPath.removeAll()
+                
+                fetchDataFromJoyDoc()
+                
+                componentTableView.delegate = viewForDataSource as? any UITableViewDelegate
+                componentTableView.dataSource = viewForDataSource as? any UITableViewDataSource
+                componentTableView.reloadData()
+            }
+        } catch {
+            print("Error decoding JSON: \(error)")
+        }
+        return nil
+    }
+}
+
+// MARK: - Function to get data from API
+func fetchDataFromJoyDoc() {
+    joyDocId = joyDocStruct?.id ?? ""
+    if joyDocStruct?.files?[0].views?.count == 0 {
+        // Fetch data for primary view
+        pageCount = joyDocStruct?.files?[0].pages?.count ?? 0
+        
+        for i in 0..<pageCount {
+            joyDocPageId.append(joyDocStruct?.files?[0].pages?[i].id ?? "")
```

Comment:
Why is ?.files an array and if multiple files need to be considered why explicitly only ever grab the first file in the array with `files?[0]`

Reply:
No actually files is always single inside files pages, views, fields and fieldPosition is multiple and these are handle dynamically according to their count. It's same as React Native